library/core/src/main/java/com/google/android/exoplayer2/extractor/flv/FlvExtractor.java

```diff
@@ -110,6 +110,7 @@ public boolean sniff(ExtractorInput input) throws IOException, InterruptedExcept
     input.peekFully(scratch.data, 0, 4);
     scratch.setPosition(0);
     int dataOffset = scratch.readInt();
+    if (dataOffset > 9) dataOffset = 9;
```

Comment:
Also, do the media files in question play correctly with the workaround in place, or do they fail to play in some other less bad way (e.g. because sniff ends up returning false)?